crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -164,6 +171,9 @@ where
     payload_builder: PayloadBuilderHandle,
     /// Listeners for engine events.
     listeners: EventListeners<BeaconConsensusEngineEvent>,
+    /// Tracks the header of invalid payloads that were rejected by the engine because they're
+    /// invalid.
+    invalid_heads: InvalidHeaderCache,
```

Comment:
nit: payload is not necessarily head
```suggestion
    invalid_headers: InvalidHeaderCache,
```